BiconeEvolution/current_antenna_evo_build/XF_Loop/Evolutionary_Loop/Antenna_Performance_Metric/physicsOfResultsPUEO.py

```diff
@@ -0,0 +1,246 @@
+                    #pathExit = passTree.event.neutrino.path.exit
+                    neutrinoFlavor = passTree.event.neutrino.flavor
+                    interactionLength = passTree.event.interaction.length
+                    interactionCrossSection = passTree.event.interaction.crossSection
+                    interactionStrength = passTree.event.interaction.strength
+                    showerPnuEv = passTree.event.shower.pnu.eV
+                    maxEField = passTree.event.signalSummaryAtDetector.maxEField
+                    maxEFieldFreq = passTree.event.signalSummaryAtDetector.maxEFieldFreq
+                    if(passTree.event.neutrino.flavor==1):
+                        nu_e = 1
+                        nu_m = 0
+                        nu_t = 0
+                    elif(passTree.event.neutrino.flavor==2):
+                        nu_e = 0
+                        nu_m = 1
+                        nu_t = 0
+                    else:
+                        nu_e = 0
+                        nu_m = 0
+                        nu_t = 1
+                    
+                    all_var = [trigg, passed, passWeight, rawWeights,
+                               position, neutrinoFlavor, 
+                               interactionLength, interactionCrossSection, 
+                               interactionStrength, showerPnuEv, maxEField,
+                               maxEFieldFreq, nu_e, nu_m, nu_t]
+                    
+                    for k in range(1, len(all_var)):
+                        var_dict['{0}'.format(var[k])].append(all_var[k])
+                    
+                PassingEvents[energy].append(np.sum(nuPasses))
+                PassingWeights[energy].append(np.sum(nuWeights))
+    print("Number of events: {}".format(np.sum(TotalEvents[energy])))
+    print("Number of passing events: {}".format(np.sum(PassingEvents[energy])))
+    print("Done collecting variables")
+
+getFiles(g.source, g.energy, g.indiv)
+
+
+### Plotting ###
+mpl.rcParams['text.usetex'] = True
+# colorblind friendly colors
+plotting_colors = ['#00429d', '#4771b2', '#73a2c6', '#a5d5d8', '#ffffe0', '#ffbcaf', '#f4777f', '#cf3759', '#93003a']
+
+#plotting the number of each type of neutrino that passed
+#Note: 1 = nu_e, 2 = nu_mu, 3 = nu_tau
+fig1 = plt.figure()
+ax1 = fig1.add_subplot(111)
+N, bins, patches = ax1.hist(var_dict['neutrinoFlavor'], bins=3, range=(0.5, 3.5), align='mid', rwidth=0.8)
+patches[0].set_facecolor(plotting_colors[0])
+patches[1].set_facecolor(plotting_colors[4])
+patches[2].set_facecolor(plotting_colors[-1])
+ax1.set_xlabel('Neutrino Flavor')
+ax1.set_ylabel('Number of Neutrinos')
+ax1.set_title('Neutrino Flavor for {} EeV Neutrinos'.format(g.energy))
+ax1.set_xticks([1, 2, 3])
+ax1.set_xticklabels(['$\\nu_e$', '$\\nu_\\mu$', '$\\nu_\\tau$'])
+ax1.grid(True)
+fig1.savefig('{}/{}_neutrinoFlavor_bestindiv.png'.format(g.destination, g.indiv))
```

Comment:
Here you can use f-strings too.
```suggestion
ax1.set_title(f'Neutrino Flavor for {g.energy} EeV Neutrinos')
ax1.set_xticks([1, 2, 3])
ax1.set_xticklabels(['$\\nu_e$', '$\\nu_\\mu$', '$\\nu_\\tau$'])
ax1.grid(True)
fig1.savefig(f'{g.destination}/{g.indiv}_neutrinoFlavor_bestindiv.png')
```
The same can be done on the rest of the file.